qemu/tests/mq_change_qnum.py

```diff
@@ -33,26 +33,37 @@ def change_queues_number(session, ifname, q_number, queues_status=None):
         if not queues_status:
             queues_status = get_queues_status(session, ifname)
 
-        if (q_number != queues_status[1] and q_number <= queues_status[0] and
-                q_number > 0):
-            expect_status = 0
-        else:
-            expect_status = 1
-
         status, output = session.cmd_status_output(mq_set_cmd, safe=True)
         cur_queues_status = get_queues_status(session, ifname)
-        if status != expect_status:
-            err_msg = "Change queues number failed, "
-            err_msg += "current queues set is %s, " % queues_status[1]
-            err_msg += "max allow queues set is %s, " % queues_status[0]
-            err_msg += "when run cmd: '%s', " % mq_set_cmd
-            err_msg += "expect exit status is: %s, " % expect_status
-            err_msg += "output: '%s'" % output
-            test.fail(err_msg)
-        if not status and cur_queues_status == queues_status:
-            test.fail("params is right, but change queues failed")
-        elif status and cur_queues_status != queues_status:
-            test.fail("No need change queues number")
+
+        if q_number == queues_status[1]:
+            if cur_queues_status != queues_status:
+                err_msg = "The queues number is wrong, "
+                err_msg += ("we are expecting [%s][%s], "
+                            % (queues_status[0], queues_status[1]))
+                err_msg += ("but current values is [%s][%s]"
+                            % (cur_queues_status[0], cur_queues_status[1]))
+                test.fail(err_msg)
+        else:
+            if q_number <= queues_status[0] and q_number > 0:
+                expect_status = 0
+            else:
+                expect_status = 1
+            if status != expect_status:
+                err_msg = "Change queues number failed, "
```

Comment:
The error msg branch below is quite vague.
For invalid param, prompt that the param is invalid, max value is xx.
For valid param, prompt set value failed. expecting [][], but the current value is [][].
Better to put the result comparison in line 48 - 51.